File: Common/windows/luacode/framework/cs_interface/uieventhelper.lua
--常用事件
--Button
--PopList
--ScrollView

local CS_UIEventHelper = CS.UIEventHelper

--设置按钮事件   Button  Button对象，Function 回调
function SetButtonClick(button, func)
    CS_UIEventHelper.SetButtonClick(button, func)
end

--设置GameObject的激活与否
function SetGoSetActive(GameObject, path, state)
    CS_UIEventHelper.SetGoSetActive(GameObject, path, state)
end

File: UnityProject-Gy/Assets/Scripts/Tools/UIEventHelper.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.Events;
using UnityEngine.UI;
using System;

public class UIEventHelper
{
    public static void SetButtonClick(Button button, UnityAction func)
    {
        button.onClick.RemoveAllListeners();
        if (func != null)
        {
            button.onClick.AddListener(func);
        }
    }

    public static void SetGoSetActive(GameObject selectGo, string path, bool state)
    {
        selectGo.transform.Find(path).gameObject.SetActive(state);
    }
}